src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy

```diff
@@ -98,17 +104,20 @@ class ComposeUp extends DefaultTask {
     }
 
     protected Iterable<ServiceInfo> loadServicesInfo() {
-        getServiceNames().collect { createServiceInfo(it) }
+        getServiceNames().collect { createServiceInfo(it) }.collectMany { [it].flatten() }
+        
     }
 
-    protected ServiceInfo createServiceInfo(String serviceName) {
-        String containerId = getContainerId(serviceName)
-        logger.info("Container ID of service $serviceName is $containerId")
-        def inspection = getDockerInspection(containerId)
-        ServiceHost host = getServiceHost(serviceName, inspection)
-        logger.info("Will use $host as host of service $serviceName")
-        def tcpPorts = getTcpPortsMapping(serviceName, inspection, host)
-        new ServiceInfo(name: serviceName, serviceHost: host, tcpPorts: tcpPorts, containerHostname: inspection.Config.Hostname, inspection: inspection)
+    protected Iterable<ServiceInfo> createServiceInfo(String serviceName) {
+        Iterable<String> containerIds = getContainerIds(serviceName)
+        containerIds.collect { String containerId ->
+            logger.info("Container ID of service $serviceName is $containerId")
+            def inspection = getDockerInspection(containerId)
+            ServiceHost host = getServiceHost(serviceName, inspection)
+            logger.info("Will use $host as host of service $serviceName")
+            def tcpPorts = getTcpPortsMapping(serviceName, inspection, host)
+            new ServiceInfo(name: inspection.Name.replace('/', ''), serviceHost: host, tcpPorts: tcpPorts, containerHostname: inspection.Config.Hostname, inspection: inspection)

```

Comment:
I would prefer to replace also the project name, like transform `test_web_1` to `web_1`. The project name is name of the directory but it can also be set by `ComposeExtension.projectName`.